contracts/BN256G1.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.5.3 <0.7.0;
+
+import "elliptic-curve-solidity/contracts/EllipticCurve.sol";
+
+
+/**
+ * @title BN256G1 Curve Library
+ * @dev Library providing arithmetic operations over G1 in bn256.
+ * Provides additional methods like pairing and pairing_batch
+ * Heavily influenced by https://github.com/PhilippSchindler/ethdkg
+ * Calls to assembly are public and not external because assembly cannot be applied on calldata
+ * @author Witnet Foundation
+ */
+
+library BN256G1 {
+
+  // Generator coordinate `x` of the EC curve
+  uint256 public constant GX = 1;
+  // Generator coordinate `y` of the EC curve
+  uint256 public constant GY = 2;
+  // Constant `a` of EC equation
+  uint256 public constant AA = 0;
+  // Constant `b` of EC equation
+  uint256 public constant BB = 3;
+  // Prime number of the curve
+  uint256 public constant PP = 0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47;
+  // Order of the curve
+  uint256 public constant NN = 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001;
+
+  /// This is 0xf1f5883e65f820d099915c908786b9d3f58714d70a38f4c22ca2bc723a70f263, the last mulitple of the modulus before 2^256
+  uint256 public constant LAST_MULTIPLE_OF_PP_LOWER_THAN_2_256 = 0xf1f5883e65f820d099915c908786b9d3f58714d70a38f4c22ca2bc723a70f263;
```

Comment:
To what extent is it the `public` visibility needed for these constants?

Reply:
None, to be honest

Reply:
Just GX and GY, which might be used to calculate e(G1,P2) = e(P1, G2)